p2p/options.go

```diff
@@ -168,3 +170,13 @@ func WithChainID[T ClientParameters](chainID string) Option[T] {
 		}
 	}
 }
+
+// TODO @renaynay
+func WithPeerIDStore[T ClientParameters](pidstore PeerIDStore) Option[T] {
+	return func(p *T) {
+		switch t := any(p).(type) { //nolint:gocritic
+		case *ClientParameters:
+			t.pidstore = pidstore
+		}
+	}
```

Comment:
Every time I see this I hate this even more

Reply:
no actionable here unfortunately unless you want to break constructor. I also am not a fan which is why I exported peer tracker in the first place.

Reply:
I made an issue